spec/lib/stripe/credit_card_remover_spec.rb

```diff
@@ -4,44 +4,102 @@
 require 'stripe/credit_card_remover'
 
 describe Stripe::CreditCardRemover do
-  let(:credit_card) { double('credit_card', gateway_customer_profile_id: 1) }
+  let!(:secret) { ENV.fetch('STRIPE_SECRET_TEST_API_KEY', nil) }
 
-  context 'Stripe customer exists' do
-    context 'and is not deleted' do
-      it 'deletes the credit card clone and the customer' do
-        customer = double('customer', deleted?: false)
-        allow(Stripe::Customer).to receive(:retrieve).and_return(customer)
+  let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id, user:) }
 
-        expect_any_instance_of(Stripe::CreditCardCloneDestroyer).to receive(:destroy_clones).with(
-          credit_card
-        )
-        expect(customer).to receive(:delete)
-        Stripe::CreditCardRemover.new(credit_card).call
-      end
+  let!(:user) { create(:user, email: "apple.customer@example.com") }
+  let!(:enterprise) { create(:enterprise) }
+
+  describe "#remove", :vcr, :stripe_version do
+    before { Stripe.api_key = secret }
+
+    let(:stripe_account_id) { ENV.fetch('STRIPE_ACCOUNT', nil) }
+
+    let(:stripe_account) {
+      create(:stripe_account, enterprise:, stripe_user_id: stripe_account_id)
+    }
+
+    let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id, user:) }
+
+    let(:pm_card) {
+      Stripe::PaymentMethod.create(
+        {
+          type: 'card',
+          card: {
+            number: '4242424242424242',
+            exp_month: 8,
+            exp_year: Time.zone.now.year.next,
+            cvc: '314',
+          },
+        },
+      )
+    }
+
+    let(:connected_account) do
+      Stripe::Account.create({
+                               type: 'standard',
+                               country: 'AU',
+                               email: 'apple.producer@example.com'
+                             })
```

Comment:
FYI, you can omit the braces `{}` here, because Ruby is magic ✨ 
```suggestion
      Stripe::Account.create(
                               type: 'standard',
                               country: 'AU',
                               email: 'apple.producer@example.com'
                             )
```